_rules/descriptive-labels-f0c5c5.md

```diff
@@ -0,0 +1,280 @@
+---
+id: f0c5c5
+name: Form field label is descriptive
+rule_type: atomic
+description: |
+  This rule checks that labels describe the purpose of form field elements.
+accessibility_requirements: 
+  wcag21:2.4.6: # Headings and labels (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Anne Thyme Nørregaard
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+---
+
+## Applicability
+
+This rule applies to any element that:
+- has one of the following [semantic roles](#semantic-role): `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`, and
+- is [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree), and 
+- has an [accessible name](#accessible-name) or is the [labeled control](https://www.w3.org/TR/html/sec-forms.html#labeled-control) of a [visible](#visible) `label` element.
+
+**Note**: The list of applicable [semantic roles](#semantic-roles) is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
+- inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+**Note:** The `option` role is not part of the list of applicable roles, because it has a required context role that inherits from the `select` role. Furthermore, `option` does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an accessible name.
+
+**Note**: This rule is a partial check for WCAG 2.1 success criterion 2.4.6, which applies to all labels. "Label" in WCAG is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within web content.
+
+## Expectation 1
+
+The [visible](#visible) `label` element, if there is any, describes the purpose of the associated form field element. The [visible](#visible) [context](#context) of the element can be used to differentiate the purpose from other form fields on the same page.
+
+**Note:** In this case, [context](#context) can be created by headings, fieldsets and legends, text that is near the control, etc. Only [visible](#visible) elements should be considered as context for Expectation 1.
+
+**Note**: Labels do not need to be lengthy. A word, or even a single character, may suffice.
+
+## Expectation 2
```

Comment:
I agree that this rule will be simpler and easier if split in two.

---

> This rule is testing two different conditions, the visible label not matching the visible context, and the accessible name not matching the programatic context.

For the sake of nitpicking 😈 I would argue that ACT-Rules format does not define "condition" (and even puts in in quotation marks which seems to be a way to indicate a blurry term if I judge by the previous sentence of ` what elements, nodes or other "parts" of a test subject are to be tested`) So it's never completely clear what is one or two conditions.

I would further argue that some of the other rules with several expectations are actually testing several conditions, *e.g.*:
* [Video as a media alternative for text](https://act-rules.github.io/rules/ab4d13) (1. the info in the video is contained somewhere else; 2. the video is labelled as an alternative for something (not necessarily the same as in 1.))
* [HTML Page has a title](https://act-rules.github.io/rules/2779a5) (1. the page has at least one `title`; 2. the first `title` has text content)
* …
(and yet these rules would be needlessly complicated if split in two, while this rule would be simplified…)

(as I rule of thumb, I would argue that if a rule has several expectation that do not depend on each other, then it's probably 2 conditions rather than one…)